cdap-app-fabric/src/main/java/co/cask/cdap/metadata/MetadataHttpHandler.java

```diff
@@ -343,13 +340,11 @@ private MetadataEntity getMetadataEntityFromPath(String uri, @Nullable  String e
   @VisibleForTesting
   static MetadataEntity.Builder makeBackwardCompatible(MetadataEntity.Builder builder) {
     MetadataEntity entity = builder.build();
-    Iterator<MetadataEntity.KeyValue> iterator = entity.iterator();
-    List<MetadataEntity.KeyValue> entityKeyValues = StreamSupport.stream(
-      Spliterators.spliteratorUnknownSize(iterator, Spliterator.ORDERED), false).collect(Collectors.toList());
-    if (entityKeyValues.size() == 3 &&
+    List<MetadataEntity.KeyValue> entityKeyValues = StreamSupport.stream(entity.spliterator(),
+                                                                         false).collect(Collectors.toList());
```

Comment:
nit: easier to read if 'false' is on the previous line with only the collect call on this line.

Reply:
Done.